### DISCUSSION TITLE
How to obtain a secret when using docker?

### DISCUSSION THREAD

**Discussion by lesha-co** [9seconds/mtg#250] [Q&A] [answered]
I'm reading a readme now and it says:

```
$ mtg generate-secret --hex google.com
ee473ce5d4958eb5f968c87680a23854a0676f6f676c652e636f6d
```

this implies that I need to attach a shell to container and run `mtg` from there right? 

I run `docker run -d --name mtg mtg`, it starts, spits configuration in logs and after some time, silently terminates:

```
{
  "level": "debug",
  "configuration": {
    // whatever 
  },
  "logger": "",
  "timestamp": 1646386068599,
  "message": "configuration"
}
```

How am I supposed to run `mtg generate-secret`?

I tried running `docker attach mtg`, it just hangs and doesn't do anything

**Comment by themegabyte** [comment]
Hey @lesha-co,

Have you tried `docker exec -it <container_PID> /mtg generate-secret -hex google.com`?

You can get the PID of a running container with: `docker ps`.

Let me know if it helped.

**Comment by jmluang** [comment, marked as answer]
if you just want to generate a secret, try this

` docker run --rm nineseconds/mtg generate-secret --hex google.com `

**Comment by 9seconds** [comment]
> this implies that I need to attach a shell to container and run mtg from there right?

Not really. You can download a binary and generate this secret or you can check other instructions available everywhere. These secrets are not something unique for mtg. You can use, for example, [seriyps.ru/mtpgen.html](https://seriyps.ru/mtpgen.html) to generate one.

Or you can use mentioned instructions, yes. Thanks, @themegabyte and @jmluang
